### PR TITLE
Parse Multi Value Query String back to array

### PR DESCRIPTION
**Pull request by nandogameiro**
Currently Bref layer doesn’t stream down the parameter to laravel when we send a request with query string parameter such as `date=xxx&date=yyy`.

This solution implements a fix at Bref layer package to parse `date=xxx&date=yyy` back to `date[]=xxx&date[]=yyy` so at laravel level we can do the magic to convert it back to `date=xxx&date=yyy`

### CODE REVIEW

File: tests/Event/Http/CommonHttpTest.php [hunk 1]
```diff
@@ -100,13 +100,14 @@ public function test request with multivalues query string have basic su
...
             'foo' => ['bar', 'baz'],
+            'date' => ['2021-07-06', '2021-08-06'],
             'cards' => ['birthday'],
             'colors' => [['red'], ['blue']],
             'shapes' => ['a' => ['square', 'triangle']],
             'myvar' => 'abc',
         ]);
-        $this->assertQueryString('foo%5B0%5D=bar&foo%5B1%5D=baz&cards%5B0%5D=birthday&colors%5B0%5D%5B0%5D=red&colors%5B1%5D%5B0%5D=blue&shapes%5Ba%5D%5B0%5D=square&shapes%5Ba%5D%5B1%5D=triangle&myvar=abc');
-        $this->assertUri('/path?foo%5B0%5D=bar&foo%5B1%5D=baz&cards%5B0%5D=birthday&colors%5B0%5D%5B0%5D=red&colors%5B1%5D%5B0%5D=blue&shapes%5Ba%5D%5B0%5D=square&shapes%5Ba%5D%5B1%5D=triangle&myvar=abc');
+        $this->assertQueryString('foo%5B0%5D=bar&foo%5B1%5D=baz&date%5B0%5D=2021-07-06&date%5B1%5D=2021-08-06&cards%5B0%5D=birthday&colors%5B0%5D%5B0%5D=red&colors%5B1%5D%5B0%5D=blue&shapes%5Ba%5D%5B0%5D=square&shapes%5Ba%5D%5B1%5D=triangle&myvar=abc');
```
Comment (karl-d) [inline tests/Event/Http/CommonHttpTest.php:111]: Are we expecing this to have the `data[0]` and `date[1]` as part of the url now?

**Review by karl-d — APPROVED** [review]
Looking all good, just one query around the `[]` in the string

File: tests/Event/Http/CommonHttpTest.php (hunk 1 above)
Comment (nandogameiro) [reply tests/Event/Http/CommonHttpTest.php:111]: we send `date=xxx&date=yyy` and bref layer converts to `date[0]=xxx&date[1]=yyy` then send it down to laravel which will handle it converting back to `date=xxx&date=yyy`

**Comment by nandogameiro** [comment]
rebased at v1.5.7

**Comment by nandogameiro** [comment]
rebased at v1.7.9

**Pull request merged**